src/js/components/DateInput/DateInput.js

```diff
@@ -396,7 +395,14 @@ Use the icon prop instead.`,
     if (open && !readOnly) {
       return [
         input,
-        <Keyboard key="drop" onEsc={() => ref.current.focus()}>
+        <Keyboard
+          key="drop"
+          onEsc={() => {
+            setTimeout(() => {
+              ref?.current.focus();
+            }, 1);
```

Comment:
Generally it feels like we try to avoid timeouts like this. Is the calendar capturing the Esc event? or placing focus elsewhere in response?

Something feels a little fishy here.

Reply:
From what I understand when we press the escape key the drop closes causing a re-render. At the same time the onEsc function gets triggered. If we try to do `ref?.current.focus();`, ref.current won't be defined yet because of the re-render caused by the drop closing. However if we add a timeout, then ref.current will be defined and we can set focus back on the DropButton.

Basically just moved what was on line 244 to line 401

Reply:
I see, thanks for clarifying! Hadn't realized the logic was also already part of the component.